airflow/models/taskinstance.py

```diff
@@ -504,6 +791,34 @@ def _execute_callable(context: Context, **execute_callable_kwargs):
     return result
 
 
+def _set_ti_attrs(target, source):
```

Comment:
Can you add some typing?
```suggestion
def _set_ti_attrs(target: TaskInstance | TaskInstancePydantic, source: TaskInstance | TaskInstancePydantic):
```